build.sbt

```diff
@@ -1095,6 +1094,31 @@ lazy val bench = http4sProject("bench")
   )
   .dependsOn(core.jvm, circe.jvm, emberCore.jvm)
 
+lazy val jsArtifactSizeTest = http4sProject("js-artifact-size-test")
+  .enablePlugins(ScalaJSPlugin, NoPublishPlugin)
+  .settings(
+    scalaJSUseMainModuleInitializer := true,
+    Test / test := {
+      val log = streams.value.log
+      val file = (Compile / fullOptJS).value.data
+      val size = io.Using.fileInputStream(file) { in =>
+        var size = 0L
+        IO.gzip(in, _ => size += 1)
+        size
+      }
+      val sizeKB = size / 1000
+      // not a hard target. increase *moderately* if need be
+      // linking MimeDB results in a 100 KB increase. don't let that happen :)
+      val targetKB = 350
+      val msg = s"fullOptJS+gzip generated ${sizeKB} KB artifact (target: <$targetKB KB)"
+      if (sizeKB < targetKB)
+        log.info(msg)
+      else
+        sys.error(msg)
```

Comment:
Turned my little example app into a proper test. This one specifically tracks the gzip size (we are currently clocking in around 325 KB). Hopefully it won't become annoying since dependency updates can also affect this.